src/main/java/com/hp/posts/controller/PostController.java

```diff
@@ -0,0 +1,57 @@
+package com.hp.posts.controller;
+
+import antlr.StringUtils;
+import com.hp.posts.model.PostModel;
+import com.hp.posts.repository.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class PostController {
+
+    @Autowired
+    private PostRepository postRepository;
+
+    @GetMapping("/posts")
+    public ResponseEntity<List<PostModel>> getAll() {
+        return ResponseEntity.ok(postRepository.findAll());
+    }
+
+    @PostMapping("/upvote/{postId}")
+    public ResponseEntity<?> increaseVote(@PathVariable Integer postId) {
+        Optional<PostModel> optionalPostModel = postRepository.findById(postId);
+        if (optionalPostModel.isPresent()) {
+            PostModel postModel = optionalPostModel.get();
+            postModel.setVotesUp(postModel.getVotesUp()+1);
+            postModel.setVotesUpPercentage(postModel.getVotesUp().toString().substring(0, 1).concat("%"));
```

Comment:
Could you please explain this line(line 34)?

Reply:
i am extracting the first digit of the vote, to build the percentage, as it is asked

Reply:
Hi Hugo. Humm, it seems you missunderstood the challenge. Check it out once more please:

"Your solution should consider that if you have a first post with 600 up votes and 400 down votes means that you have 60% of up votes and 40% of down votes, and if you have another post with 6 up votes and 4 down votes you have the same % of the previous post, 60% of up votes and 40% of down votes.
Hint: Note that the score in % of the two posts are equal, but the real "values" are significantly differents"

Your solution should be able to distinguish the weight of the total votes for each post, and a post with 1000 votes and 60% upvote is more important than a post with 10 votes with 60% upvote.

Reply:
hi, fixed. if you could gave a look, much thanks